### PR TITLE
🌑 Integrate with Gitcoin backend

### PR DESCRIPTION
**Pull request by pazernykormoran**
The only problematic thing is that IDK when to display screen "You don't have Gitcoin passport". When someone has no passport his score is returned normally as 0. For now this screen is never displayed.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207203989412399
  - https://app.asana.com/0/0/1207167854038957

### CODE REVIEW

File: packages/frontend/src/components/userActious/gitcoin/CheckingGitcoinScore.tsx [hunk 1]
```diff
@@ -31,7 +37,34 @@ const gitcoinScoreSteps = [
...
+    queryKey: ['gitcoinScore', address, chainId],
+    queryFn: () => getGitcoinScore(address as Hex, chainId),
+    enabled: !!(requestSettled && address),
+    retry: true,
+    retryDelay: 5000,
+  })
+  const step = requestSettled ? 2 : 1
+
+  useEffect(() => {
+    mutate()
```
Comment (b-tarczynski) [inline packages/frontend/src/components/userActious/gitcoin/CheckingGitcoinScore.tsx:59]: I believe that we can have some optional onChange prop in step that can be invoked when we change step

File: packages/frontend/src/backend/getPassportScore.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { useMutation } from '@tanstack/react-query'
+import { useState } from 'react'
+import { Hex } from 'viem'
+import { useAccount, useChainId, useSignMessage } from 'wagmi'
+
+export const useSendForScoring = () => {
+  const { address } = useAccount()
+  const chainId = useChainId()
+  const { signMessageAsync } = useSignMessage()
+  const [requestSettled, setRequestSettled] = useState(false)
```
[the remark this answers is not in the document]
Comment (pazernykormoran) [reply packages/frontend/src/backend/getPassportScore.ts:17]: Done

File: packages/frontend/src/backend/getPassportScore.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      mutationFn: async () => {
+        if (requestSettled || !address) {
+          return
+        }
+
+        try {
+          await getGitcoinScore(address, chainId)
+          setRequestSettled(true)
+          return
+        } catch (error) {}
```
[the remark this answers is not in the document]
Comment (pazernykormoran) [reply packages/frontend/src/backend/getPassportScore.ts:30]: We are sending address for scoring only if this request fails.

File: packages/frontend/src/components/userActious/gitcoin/CheckingGitcoinScore.tsx (hunk 1 above)
Comment (pazernykormoran) [reply packages/frontend/src/components/userActious/gitcoin/CheckingGitcoinScore.tsx:59]: I have refactored this logic so there is no more useEffect

**Review by p-sad — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/userActious/gitcoin/CheckingGitcoinScore.tsx [hunk 4]
```diff
@@ -31,7 +37,37 @@ const gitcoinScoreSteps = [
...
+    retry: true,
+    retryDelay: 5000,
+  })
+  const step = gitcoinRequestSettled ? 2 : 1
+
+  useEffect(() => {
+    if (data && data.status == 'done') {
+      setGitcoinCredentials(data)
+    }
+  }, [data, setGitcoinCredentials])
```
[the remark this answers is not in the document]
Comment (p-sad) [reply packages/frontend/src/components/userActious/gitcoin/CheckingGitcoinScore.tsx:69]: Nevermind, you shouldn't do `onSuccess` on queries

**Review by p-sad — APPROVED** [review]

**Pull request merged**